persistence/trees/trees_test.go

```diff
@@ -1,111 +1,205 @@
+		context := newTestPostgresContext(t, 0, pmod)
+
+		err := context.RollbackToSavePoint()
+		require.Error(t, err)
+		require.ErrorIs(t, err, trees.ErrFailedRollback)
+	})
 }
 
-func TestTreeStore_New(t *testing.T) {
-	// TODO: Write test case for the NewStateTrees function
-	t.Skip("TODO: Write test case for NewStateTrees function")
+func newTestPersistenceModule(t *testing.T, databaseURL string) modules.PersistenceModule {
+	t.Helper()
+	teardownDeterministicKeygen := keygen.GetInstance().SetSeed(42)
+	defer teardownDeterministicKeygen()
+
+	cfg := newTestDefaultConfig(t, databaseURL)
+	genesisState, _ := test_artifacts.NewGenesisState(
+		genesisStateNumValidators,
+		genesisStateNumServicers,
+		genesisStateNumApplications,
+		genesisStateNumServicers,
+	)
+
+	runtimeMgr := runtime.NewManager(cfg, genesisState)
+
+	bus, err := runtime.CreateBus(runtimeMgr)
+	require.NoError(t, err)
+
+	persistenceMod, err := persistence.Create(bus)
+	require.NoError(t, err)
+
+	return persistenceMod.(modules.PersistenceModule)
 }
 
-func TestTreeStore_DebugClearAll(t *testing.T) {
-	// TODO: Write test case for the DebugClearAll method
-	t.Skip("TODO: Write test case for DebugClearAll method")
+// fetches a new default node configuration for testing
+func newTestDefaultConfig(t *testing.T, databaseURL string) *configs.Config {
+	t.Helper()
+	cfg := &configs.Config{
+		Persistence: &configs.PersistenceConfig{
+			PostgresUrl:       databaseURL,
+			NodeSchema:        testSchema,
+			BlockStorePath:    ":memory:",
+			TxIndexerPath:     ":memory:",
+			TreesStoreDir:     ":memory:",
+			MaxConnsCount:     5,
+			MinConnsCount:     1,
+			MaxConnLifetime:   "5m",
+			MaxConnIdleTime:   "1m",
+			HealthCheckPeriod: "30s",
+		},
+	}
+	return cfg
 }
+func createAndInsertDefaultTestApp(t *testing.T, db *persistence.PostgresContext) (*core_types.Actor, error) {
+	t.Helper()
+	app := newTestApp(t)
 
-// TODO_AFTER(#861): Implement this test with the test suite available in #861
-func TestTreeStore_GetTreeHashes(t *testing.T) {
-	t.Skip("TODO: Write test case for GetTreeHashes method") // context: https://github.com/pokt-network/pocket/pull/915#discussion_r1267313664
+	addrBz, err := hex.DecodeString(app.Address)
+	require.NoError(t, err)
+
+	pubKeyBz, err := hex.DecodeString(app.PublicKey)
+	require.NoError(t, err)
+
+	outputBz, err := hex.DecodeString(app.Output)
+	require.NoError(t, err)
+	return app, db.InsertApp(
+		addrBz,
+		pubKeyBz,
+		outputBz,
+		false,
+		defaultStakeStatus,
+		defaultStake,
+		defaultChains,
+		defaultPauseHeight,
+		defaultUnstakingHeight)
 }
 
-func TestTreeStore_Prove(t *testing.T) {
-	nodeStore := kvstore.NewMemKVStore()
-	tree := smt.NewSparseMerkleTree(nodeStore, smtTreeHasher)
-	testTree := &stateTree{
-		name:      "test",
-		tree:      tree,
-		nodeStore: nodeStore,
-	}
+// TECHDEBT(#796): Test helpers should be consolidated in a single place
+func newTestApp(t *testing.T) *core_types.Actor {
```

Comment:
OPTIONAL: Per @h5law's comment, we could move the helpers into a separate `utils_test.go` but don't feel strongly about it.